File: frontend/src/models/wallpaper.ts
import { Category } from "./category";

export interface Tag {
  id: number;
  name: string;
}

export interface Wallpaper {
  id: number;
  image_url: string;
  image_thumb_url: string;
  image_medium_url?: string;
  category: Category;
  tags?: Tag[];
  created_at: string;
  updated_at: string;
}

export interface WallpaperResponse {
  wallpapers: Wallpaper[];
  total: number;
  limit: number;
  offset: number;
}

export interface NextPreviousWallpaperFilter {
  category?: string;
  search?: string;
}

File: server/internal/domain/models/dto/wallpaper.go
package dto

import (
	"wallpaperio/server/internal/domain/models"
)

type Direction string

const (
	DirectionNext     Direction = "next"
	DirectionPrevious Direction = "previous"
)

type CreateWallpaper struct {
	ImageURL       string   `json:"image_url"`
	ImageThumbUrl  string   `json:"image_thumb_url"`
	ImageMediumUrl *string  `json:"image_medium_url,omitempty"`
	Category       string   `json:"category"`
	Tags           []string `json:"tags"`
	Prompt         string   `json:"prompt"`
	Width          int      `json:"width"`
	Height         int      `json:"height"`
}

type WallpaperFilter struct {
	Tags     []string
	Category string
	Search   string
	Limit    int
	Offset   int
}

type NextPreviousWallpaperFilter struct {
	Category  string
	Search    string
	CurrentID uint64
}

type WallpaperResult struct {
	Wallpapers []models.Wallpaper
	Total      int64
}

type FindSimilarWallpapers struct {
	Features  []float32
	Limit     int
	Offset    *int64
	ExcludeID *uint
}

type GetSimilarWallpapersRequest struct {
	Key    string `json:"key" form:"key" binding:"required"`
	Limit  int    `json:"limit" form:"limit"`
	Offset int    `json:"offset" form:"offset"`
}
